config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +115,200 @@ const (
+	// configure the persistent volume claim, including storage class, volume
+	// size, and name.
+	// If omitted, the Pod uses ephemeral storage and alert data will not persist
+	// across restarts.
+	// // This field is optional.
+	// +optional
+	VolumeClaimTemplate *v1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
+}
+
+// AlertManagerDeployMode defines the deployment state of the platform Alertmanager instance.
+//
+// Possible values:
+// - "Disabled": The Alertmanager instance will not be deployed.
+// - "DefaultConfig": The Alertmanager instance will be deployed with default settings.
+// - "CustomConfig": The Alertmanager instance will be deployed with custom configuration.
+type AlertManagerDeployMode string
+
+const (
+	// AlertManagerModeDisabled means the Alertmanager instance will not be deployed.
+	AlertManagerDeployModeDisabled AlertManagerDeployMode = "Disabled"
+	// AlertManagerModeDefaultConfig means the Alertmanager instance will be deployed with default settings.
+	AlertManagerDeployModeDefaultConfig AlertManagerDeployMode = "DefaultConfig"
+	// AlertManagerModeCustomConfig means the Alertmanager instance will be deployed with custom configuration.
+	AlertManagerDeployModeCustomConfig AlertManagerDeployMode = "CustomConfig"
+)
+
+// logLevel defines the verbosity of logs emitted by Alertmanager.
+// Valid values are Error, Warn, Info and Debug.
+// +kubebuilder:validation:Enum=Error;Warn;Info;Debug
+type LogLevel string
+
+const (
+	// Error only errors will be logged.
+	LogLevelError LogLevel = "Error"
+	// Warn, both warnings and errors will be logged.
+	LogLevelWarn LogLevel = "Warn"
+	// Info, general information, warnings, and errors will all be logged.
+	LogLevelInfo LogLevel = "Info"
+	// Debug, detailed debugging information will be logged.
+	LogLevelDebug LogLevel = "Debug"
+)
+
+// ContainerResource defines a single resource requirement for a container.
+// +kubebuilder:validation:XValidation:rule="has(self.request) || has(self.limit)",message="at least one of request or limit must be set"
+type ContainerResource struct {
+	// name of the resource (e.g. "cpu", "memory", "hugepages-2Mi").
+	// This field is required.
+	// +required
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.qualifiedName().validate(self).hasValue()",message="name must be a valid qualified name"
+	Name string `json:"name"`
+
+	// request is the minimum amount of the resource required (e.g. "2Mi", "1Gi").
+	// This field is optional.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="isQuantity(self) && quantity(self).compareTo(quantity('0')) >= 0",message="request must be a non-negative quantity"
+	// +kubebuilder:validation:XValidation:rule="(has(self.request) && has(self.limit) && isQuantity(self.request) && isQuantity(self.limit)) ? quantity(self.limit).compareTo(quantity(self.request)) >= 0 : true",message="limit should not be less than request"
```

Comment:
This is in the wrong place, it needs to be on the `ContainerResource` struct

Reply:
done, txy